galley/pkg/config/analysis/analyzer.go

```diff
@@ -14,41 +14,89 @@
 
 package analysis
 
-import "istio.io/istio/galley/pkg/config/scope"
+import (
+	"istio.io/istio/galley/pkg/config/collection"
+	"istio.io/istio/galley/pkg/config/scope"
+)
 
 // Analyzer is an interface for analyzing configuration.
 type Analyzer interface {
-	Name() string
+	Metadata() Metadata
 	Analyze(c Context)
 }
 
+// Metadata represents metadata for an analyzer
+type Metadata struct {
+	name            string
+	collectionsUsed map[collection.Name]bool
+}
+
+// NewMetadata creates a new Metadata object
+func NewMetadata(name string, collectionsUsed collection.Names) Metadata {
+	cu := make(map[collection.Name]bool)
+	for _, c := range collectionsUsed {
+		cu[c] = true
+	}
+	return Metadata{
+		name:            name,
+		collectionsUsed: cu,
+	}
+}
+
+// Name of the analyzer this metadata is for
+func (m Metadata) Name() string {
+	return m.name
+}
+
+// CollectionsUsed registers the collections accessed by the analyzer this metadata is for
+func (m Metadata) CollectionsUsed() map[collection.Name]bool {
+	return m.collectionsUsed
+}
+
 type combinedAnalyzers struct {
-	name      string
+	metadata  Metadata
 	analyzers []Analyzer
 }
 
 // Combine multiple analyzers into a single one.
+// For collections used, use the union of the component analyzers
 func Combine(name string, analyzers ...Analyzer) Analyzer {
 	return &combinedAnalyzers{
-		name:      name,
+		metadata:  NewMetadata(name, combineCollectionsUsed(analyzers)),
 		analyzers: analyzers,
 	}
 }
 
-// Name implements Analyzer
-func (c *combinedAnalyzers) Name() string {
-	return c.name
+// Metadata implements Analyzer
+func (c *combinedAnalyzers) Metadata() Metadata {
```

Comment:
Minor nit: It looks like Metadata is copy-by-value. You probably don't need to make its members hidden in this case. (i.e. you can avoid `Name()` and simply use a `Name string` field).

Reply:
Good point. Done.